static/js/rest.js

```diff
@@ -73,21 +75,24 @@ export const makeReducer = (endpoint: Endpoint, initialState: RestState = INITIA
   };
 };
 
-export function makeFetchFunc(endpoint: Endpoint): (...args: any) => Promise<*> {
-  return (...args) => fetchJSONWithCSRF(endpoint.url, endpoint.makeOptions(...args));
+export function makeFetchFunc(endpoint: Endpoint): (params?: Object) => Promise<*> {
+  return (params = {}) => {
+    let url = endpoint.urlTemplate ? endpoint.urlTemplate(params) : endpoint.url;
+    return fetchJSONWithCSRF(url || '', endpoint.fetchOptions(params));
```

Comment:
womp womp, why flow why. I'll take a look, but yeah, sometimes you have to do that with flow...

in MM in a few places I had to do something like this:

```js
let el: HTMLElement = (document.querySelector('foo'): any);
```

because flow assigns `HTMLElement | null` as the return type of `querySelector`...

similar level of 'stupid hack' there, haha